pandas-stubs/_typing.pyi

```diff
@@ -392,74 +479,40 @@ AstypeArg: TypeAlias = (
     | DtypeObj
 )
 
-# filenames and file-like-objects
-AnyStr_cov = TypeVar("AnyStr_cov", str, bytes, covariant=True)
-AnyStr_con = TypeVar("AnyStr_con", str, bytes, contravariant=True)
-
-class BaseBuffer(Protocol):
-    @property
-    def mode(self) -> str: ...
-    def seek(self, offset: int, whence: int = ..., /) -> int: ...
-    def seekable(self) -> bool: ...
-    def tell(self) -> int: ...
-
-class ReadBuffer(BaseBuffer, Protocol[AnyStr_cov]):
-    def read(self, n: int = ..., /) -> AnyStr_cov: ...
-
-class WriteBuffer(BaseBuffer, Protocol[AnyStr_con]):
-    def write(self, b: AnyStr_con, /) -> Any: ...
-    def flush(self) -> Any: ...
-
-class ReadPickleBuffer(ReadBuffer[bytes], Protocol):
-    def readline(self) -> bytes: ...
-
-class ReadCsvBuffer(ReadBuffer[AnyStr_cov], Protocol[AnyStr_cov]):
-    def __iter__(self) -> Iterator[AnyStr_cov]: ...
-    def fileno(self) -> int: ...
-    def readline(self) -> AnyStr_cov: ...
-    @property
-    def closed(self) -> bool: ...
-
-class WriteExcelBuffer(WriteBuffer[bytes], Protocol):
-    def truncate(self, size: int | None = ..., /) -> int: ...
-
-FilePath: TypeAlias = str | PathLike[str]
+# DtypeArg specifies all allowable dtypes in a functions its dtype argument
+DtypeArg: TypeAlias = Dtype | Mapping[Hashable, Dtype]
+DtypeObj: TypeAlias = np.dtype[np.generic] | ExtensionDtype
 
-_T_co = TypeVar("_T_co", covariant=True)
+# converters
+ConvertersArg: TypeAlias = dict[Hashable, Callable[[Dtype], Dtype]]
```

Comment:
I think this might need to be `Mapping[Hashable, Callable[[Dtype], Dtype]]` or `dict[HashableT, Callable[[Dtype], Dtype]]`  to allow sub-classes when this is used outside of bidirectional type inference (assigned to a variable and then used as an argument).

I believe we discussed Mapping+Hashable vs dict+HasableT at somepoint. I forgot which one we ended up with (feel like dict as pandas did not always accept sub-classes).

Reply:
I copied it from pandas.  Will change to `Mapping` here.